packages/ionic/src/commands/deploy/core.ts

```diff
@@ -0,0 +1,374 @@
+      return;
+    }
+    // parse it with elementtree
+    let etree;
+    try {
+      etree = et.parse(stringData);
+    } catch (e) {
+      this.env.log.error(`Impossible to parse the XML in the string.xml`);
+      this.printStringXmlInstructions(options);
+      return;
+    }
+    // check that it is an actual string.xml file (root tag is resources)
+    const root = etree.getroot();
+    if (root.tag !== 'resources') {
+      this.env.log.error(`string.xml is not a valid android string.xml file because the root is not a <resources> tag`);
+      this.printStringXmlInstructions(options);
+      return;
+    }
+    for (const [optionKey, stringKey] of Object.entries(this.optionsToStringXmlKeys)) {
+      let element = root.find(`./string[@name="${stringKey}"]`);
+      // if the tag already exists, just update the content
+      if (element) {
+        element.text = options[optionKey] as string;
+      } else {
+        // otherwise create the tag
+        element = et.SubElement(root, 'string');
+        element.set('name', stringKey);
+        element.text = options[optionKey] as string;
+      }
+    }
+    // write back the modified plist
+    const newXML = etree.write({
+      encoding: 'utf-8',
+      indent: 2,
+    });
+    try {
+      fs.writeFileSync(stringXmlPath, newXML, { encoding: 'utf-8' });
+    } catch (e) {
+      this.env.log.error(`Changes to string.xml could not be written.`);
+      this.printStringXmlInstructions(options);
+    }
+    this.env.log.ok(`cordova-plugin-ionic variables correctly added to the Android project`);
+  }
+
+  protected async preRunCheckInputs(options: CommandLineOptions): Promise<void> {
+    const updateMethodList: string[] = ['auto', 'background', 'none'];
+    const defaultUpdateMethod = 'background';
+    // handle the app-id option in case the user wants to override it
+    if (!options['app-id']) {
+      const appId = await this.getAppId();
+      if (!appId) {
+        this.env.log.warn(
+          `No app id found on the project: consider running ${input('ionic link')} to Connect local apps to Ionic`
+        );
+      }
+      const appIdOption = await this.env.prompt({
+        type: 'input',
+        name: 'app-id',
+        message: `Appflow app ID:`,
+        default: appId,
+      });
+      options['app-id'] = appIdOption;
+    }
+
+    if (!options['channel-name']) {
+      options['channel-name'] = await this.env.prompt({
+        type: 'input',
+        name: 'channel-name',
+        message: `Channel Name:`,
+        validate: v => validators.required(v),
+      });
+    }
+
+    // validate that the update-method is allowed
+    let overrideUpdateMethodChoice = false;
+    if (options['update-method'] && !updateMethodList.includes(options['update-method'] as string)) {
+      if (this.env.flags.interactive) {
+        this.env.log.nl();
+        this.env.log.warn(`${strong(options['update-method'] as string)} is not a valid update method; choose a valid one`);
```

Comment:
Please use the `input` color. update-method is a user input. 

Additionally, in other places in help documentation and messages, we prefix these options in prose with `--` so users know what we're referring to, even for prompted options.

```ts
strong(options['update-method'] as string) -> input(`--${options['update-method']}`)
```